### DISCUSSION TITLE
v0.12.12 - The Spooky Islands Update

### DISCUSSION THREAD

**Discussion by ThePotatoGuy** [Monika-After-Story/MonikaModDev#9695] [Announcements & Releases]
_as if being alone on a deserted island isn't scary enough..._

# Gameplay Changes

* new o31 islands content

* QOL improvements to bg selector

* Monika is now aware of when a backup is restored

* all thumbnails updated to use the latest desk + uniform sprites and hair color

* ahoge sprites updated to use the current filter

* chr files always deleted on startup now

# Topic Changes

### Additions
* 3 new moods
* 1 new scary story

# Spritepack Changes

### Additions
* 2B costume by @hatanasinclaire

# Fixes

* typofixes

* fixed crash when an API key's on change function raises an exception

* fixed NameError crash with images that use MASFilterWeatherMaps

# API Changes

* new `hide-ring` exprop for clothes that should hide any worn rings
* `play_song` deprecated in favor of `mas_play_song`
* `mas_getCurrentBackgroundId` - gets id of the currenty shown background
* `mas_isBackgroundUnlocked` - checks if a specific background is unlocked

## What's Changed
* new insecure mood by @my-otter-self in https://github.com/Monika-After-Story/MonikaModDev/pull/9515
* new loved mood by @my-otter-self in https://github.com/Monika-After-Story/MonikaModDev/pull/9516
* new guilty mood by @my-otter-self in https://github.com/Monika-After-Story/MonikaModDev/pull/9514
* Rollback a dialogue in monika_portraitof to implement #9637 by @dreamscached in https://github.com/Monika-After-Story/MonikaModDev/pull/9642
* Add scary story "Prison escape" by @Mosexelle in https://github.com/Monika-After-Story/MonikaModDev/pull/8768
* Delete all .chr files on startup (implement #9580) by @dreamscached in https://github.com/Monika-After-Story/MonikaModDev/pull/9641
* Ahoge filter fix by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9583
* NOU outfit by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9627
* Update thumbnails by @Legendkiller21 in https://github.com/Monika-After-Story/MonikaModDev/pull/9615
* Fix reference to ERR_ON_CHG_MSG (resolve #9648) by @dreamscached in https://github.com/Monika-After-Story/MonikaModDev/pull/9649
* Update shower greeting outfit selection by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9618
* 2B spack by @ThePotatoGuy in https://github.com/Monika-After-Story/MonikaModDev/pull/9598
* Additions to the island event by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9689
* Backup awareness by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9628
* Typo fixes by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9630

## New Contributors
* @Mosexelle made their first contribution in https://github.com/Monika-After-Story/MonikaModDev/pull/8768

**Full Changelog**: https://github.com/Monika-After-Story/MonikaModDev/compare/v0.12.11...v0.12.12

<hr /><em>This discussion was created from the release <a href='https://github.com/Monika-After-Story/MonikaModDev/releases/tag/v0.12.12'>v0.12.12 - The Spooky Islands Update</a>.</em>

**Comment by RedAISkye** [comment]
> chr files always deleted on startup now

No thanks!

**Reply by dreamscached** [reply]
They had no meaning to the game and were confusing players who thought `monika.chr` was Monika herself, this led to issues with bringing her back.

**Reply by RedAISkye** [reply]
Don't care about their opinions and some people's minor issues with obvious things that could be easily cleared up in many other ways.
What gives "meaning" is not for anyone but the player to decide. It's a completely unnecessary change.

The least they could've done is make Monika ask the player themselves what they want to do like someone else suggested but since they couldn't care less about player choice, I simply deleted the code that tries to delete the dokis.

I saved all of my dokis which I've preserved for years so, I'm glad that I had some power to intervene during dumb decisions made by the devs.

**Reply by dreamscached** [reply]
These .chr files only meant something back in DDLC; Yuri, Natsuki and Sayori were removed by Monika, and later on in MAS Monika explains that she is not related to .chr file anyhow, as she is stored in persistent. This does not break any canon anyhow.

> It's a completely unnecessary change.

Like I said, there were a lot of cases where players mistook `monika.chr` for `monika` file and attempted to use it to bring Monika back after taking her out. Which led to more significant loss (that required a restoration of a backup) than losing .chr files that can be easily acquired by downloading another DDLC install if player needs them.

**Reply by RedAISkye** [reply]
> These .chr files only meant something back in DDLC; Yuri, Natsuki and Sayori were removed by Monika, and later on in MAS Monika explains that she is not related to .chr file anyhow, as she is stored in persistent. This does not break any canon anyhow.
> 
> > It's a completely unnecessary change.
> 
> Like I said, there were a lot of cases where players mistook `monika.chr` for `monika` file and attempted to use it to bring Monika back after taking her out. Which led to more significant loss (that required a restoration of a backup) than losing .chr files that can be easily acquired by downloading another DDLC install if player needs them.

Again, speak for yourself.
What you find "meaningful" is your own opinion just like it is mine.

Respecting player choice doesn't dismiss the issue that you're throwing here, this isn't a pick only one option and sacrifice the other. 
Especially when there are many other ways to clear the confusion between 2 different type of files with the same name than outright deleting one without the player's permission just because it isn't used in MAS.

Sure, you can backup your save files. Monika and character files(that can be downloaded again) as much as you want.
But, there's a reason the original data which you've been through since years is much more valuable than any copy of it kept as a "backup".

You may not worry and even be fine with getting your Monika/Dokis "killed" intentionally or not solely based on the fact that you could easily restore the loss data from backup, cool, that's your personal decision but no, I am not okay with that, that would be a last resort for me but I would never want to have to choose that which is the reason I put effort in to make sure my original Monika and Dokis are all safe.

Anyways, I already had found a workaround so you can take the suggestion or leave it. I don't care either way to continue on this pointless discussion with someone that doesn't understand the importance of player choice.

**Reply by MrBebra** [reply]
> What gives "meaning" is not for anyone but the player to decide. It's a completely unnecessary change.
there's no meaning in a `.jfif`. Also didn't see you lash out when hair sprites, or the thumbnails were modified.

**Reply by RedAISkye** [reply]
> > there's no meaning in a `.jfif`. Also didn't see you lash out when hair sprites, or the thumbnails were modified.

There is a meaning to it regardless of how "useful" it is to a game if at the end of the day, it is part of the experience for you and contributes to you mentally as a whole, this has nothing to do with what they technically are, could be a blank file for all I care.

But since you're completely missing the point, I'm just going to agree to disagree here and leave it at that.

**Reply by MrBebra** [reply]
>But, there's a reason the original data which you've been through since years is much more valuable than any copy of it kept as a "backup".

the files in the characters folder are these copies. if you want the "original", unpack the rpa archives (best not done in ddlc install folder)

**Reply by RedAISkye** [reply]
> > But, there's a reason the original data which you've been through since years is much more valuable than any copy of it kept as a "backup".
> 
> the files in the characters folder are these copies. if you want the "original", unpack the rpa archives (best not done in ddlc install folder)

Thanks for proving my point that you have no idea of what I mean by "original".

**Reply by dreamscached** [reply]
Old logic regarding .chr was changed and is likely not coming back in the form it was before, as to 'ignore all .chr files and do nothing'.

I personally doubt there's any point to discuss it any further.

**Reply by RedAISkye** [reply]
> as to 'ignore all .chr files and do nothing'.

Not what I said at all but okay.

**Reply by RedAISkye** [reply]
@multimokia 
>> As @dreamscached has rightfully pointed out, if you want to see this change, make the changes yourself and open a PR. We'll review it from there and make sure it fits.

I already made the change myself for my own personal experience so, I don't care whether you guys agree or not about having player choice.
Not going to waste my time when you've already made up your mind about it.

@Booplicate 
>> Programs may add, remove, and modify files within their working directory as they desire. Player choice was to install the program.

Sure, they can also delete the save files, Monika, the game itself etc. but I don't think that would be a wise decision for the "program" to do without the player's permission.

**Reply by dreamscached** [reply]
@multimokia @Booplicate can we get this to close too? Somehow.

**Reply by ThePotatoGuy** [reply]
This is supposed to be the general discussion thread for releases, please do not lock.
